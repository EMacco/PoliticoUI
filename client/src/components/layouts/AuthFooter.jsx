import React from 'react';

const AuthFooter = () => {
  return (
    <footer className="user-home-footer">
      Politico Website Designed by Okwara Emmanuel &copy; 2019
    </footer>
  );
};

export default AuthFooter;
