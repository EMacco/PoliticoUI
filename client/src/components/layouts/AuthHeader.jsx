import React, { Component, Fragment } from 'react';
import NavItem from '@components/common/NavItem';
import { connect } from 'react-redux';
import { Redirect, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { logoutUser } from '@actions/authActions';

class AuthHeader extends Component {
  state = { showOverlay: false };

  toggleMenu = () => {
    this.setState({ showOverlay: !this.state.showOverlay });
  };

  signoutBtnClicked = () => {
    this.props.logoutUser(this.props.history);
  };

  render() {
    if (!this.props.auth.isAuthenticated) return <Redirect to="/" />;
    const { firstname, lastname, isadmin } = this.props.auth.user;
    const fullname = `${firstname} ${lastname}`;
    return (
      <Fragment>
        <header className="index-header">
          <h1 className="logo">
            <a href="user-home.html">POLITICO</a>
          </h1>

          <input type="checkbox" className="nav-toggle" id="nav-toggle" onClick={this.toggleMenu} />
          <nav>
            <div className="header-nav">
              <ul id="hideDashboard">
                <NavItem name="Home" path="/dashboard" />

                {isadmin ? <NavItem name="My Dashboard" path="/admin-dashboard" /> : ''}

                <NavItem name="Political Parties" path="/parties" />
                <NavItem name="Government Offices" path="/offices" />
                <NavItem name="Vote" path="/election" />
                <li>
                  <a onClick={this.signoutBtnClicked}>Log out</a>
                </li>
              </ul>
            </div>
            <div className="header-nav-big-screen">
              <ul>
                <li>
                  Welcome, <span id="welcomeUserLbl">{fullname}</span>
                </li>
              </ul>
            </div>
          </nav>
          <label htmlFor="nav-toggle" className="nav-toggle-label">
            <span />
          </label>
        </header>
        <div id="overlay" className={this.state.showOverlay ? 'overlay' : ''} />
      </Fragment>
    );
  }
}

AuthHeader.propTypes = {
  auth: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logoutUser }
)(withRouter(AuthHeader));
