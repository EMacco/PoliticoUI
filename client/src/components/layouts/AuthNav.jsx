import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import personIcon from '@base/img/person-large.png';
import { logoutUser } from '@actions/authActions';
import { updateProfilePicture } from '@actions/dashboard';

class AuthNav extends Component {
  state = {
    uploadingMessage: '',
    src: ''
  };

  signoutBtnClicked = () => {
    this.props.logoutUser(this.props.history);
  };

  profileImageChanged = event => {
    this.setState({ uploadingMessage: 'Uploading...' });

    if (event.target.files.length > 0) {
      this.setState({ src: URL.createObjectURL(event.target.files[0]) });
      this.props.updateProfilePicture(this.state.src);
    }
  };

  render() {
    const { firstname, lastname, email, phonenumber, passporturl, isadmin } = this.props.auth.user;

    const { uploadingMessage, src } = this.state;
    const profileImage = passporturl === 'undefined' ? personIcon : src === '' ? passporturl : src;

    return (
      <section id="user-profile-container">
        <div>
          <img
            src={profileImage === 'https://' ? personIcon : profileImage}
            id="profile-img-preview"
          />
          <i>
            <label id="uploadingText">{uploadingMessage}</label>
          </i>
          <input
            type="file"
            accept="image/x-png,image/gif,image/jpeg"
            className="select-btn"
            id="profile-file-upload"
            onChange={this.profileImageChanged}
          />
        </div>
        <div className="profile-description-text">
          <label>
            <span className="profile-answers" id="welcomeUserLbl2">
              {firstname} {lastname}
            </span>
          </label>
          <label>
            Email:{' '}
            <span className="profile-answers" id="welcomeEmailLbl">
              {email}
            </span>
          </label>
          <label>
            Phone:{' '}
            <span className="profile-answers" id="welcomePhoneLbl">
              {phonenumber}
            </span>
          </label>
        </div>

        <hr className="separate-profile-nav" />

        <Link to="/dashboard" id="current-side-bar">
          <div className="nav-menu" id="current-side-bar">
            My Profile
          </div>
        </Link>
        {isadmin ? (
          <Link to="/admin-dashboard" id="hideDashboardDiv">
            <div className="nav-menu">My Dashboard</div>
          </Link>
        ) : (
          ''
        )}
        <Link to="/parties">
          <div className="nav-menu">Political Parties</div>
        </Link>
        <Link to="/offices">
          <div className="nav-menu">Government Offices</div>
        </Link>
        <Link to="/elections">
          <div className="nav-menu">Vote</div>
        </Link>
        <Link to="/dashboard" onClick={this.signoutBtnClicked}>
          <div className="nav-menu">Log out</div>
        </Link>
      </section>
    );
  }
}

AuthNav.prototypes = {
  auth: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired,
  updateProfilePicture: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logoutUser, updateProfilePicture }
)(withRouter(AuthNav));
