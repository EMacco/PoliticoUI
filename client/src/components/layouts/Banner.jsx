import React from 'react';
import { Link } from 'react-router-dom';

const Banner = () => {
  return (
    <section id="index-banner">
			<div className="vertical-center">
				<h2>WELCOME TO POLITICO</h2>
				<h3>A transparent system that allows you to vote with integrity.</h3>
				<center><Link to="/register"><input type="submit" name="registerBtn" id="index-register-btn" value="Create Account" /></Link></center>
			</div>
		</section>
  )
}

export default Banner