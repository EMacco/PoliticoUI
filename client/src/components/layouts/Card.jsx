import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CardText from '../common/CardText';
import CardLink from '../common/CardLink';

const Card = ({ name, image, content, hidden }) => {
  const cardContent = content.map(details => {
    const {
      type,
      title,
      value,
      officeId,
      officeName,
      numberOfCandidates,
      hidden,
      text,
      style
    } = details;
    let design;

    if (hidden === 'true') return;
    switch (type) {
      case 'link':
        design = (
          <CardLink
            officeId={officeId}
            officeName={officeName}
            numberOfCandidates={numberOfCandidates}
            key={Math.random().toString()}
          />
        );
        break;
      case 'button':
        design = (
          <button className={style} key={Math.random().toString()}>
            {text}
          </button>
        );
        break;
      case 'span':
        design = (
          <span className={style} key={Math.random().toString()}>
            {text}
          </span>
        );
        break;
      default:
        design = <CardText title={title} value={value} key={Math.random().toString()} />;
    }

    return design;
  });

  return (
    <Fragment>
      <div
        className={
          hidden === 'true'
            ? 'individual-person-container hidden-div'
            : 'individual-person-container'
        }
      >
        <div>
          <img src={image} />
        </div>
        <div className="profile-description-text">
          <label>
            <span className="profile-answers">{name.charAt(0).toUpperCase() + name.substr(1)}</span>
          </label>

          {cardContent}
        </div>
      </div>
    </Fragment>
  );
};

Card.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  content: PropTypes.array,
  hidden: PropTypes.string
};

Card.defaultProps = {
  name: '',
  image: '',
  content: [],
  hidden: 'false'
};

export default Card;
