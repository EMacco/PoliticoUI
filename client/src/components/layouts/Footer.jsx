import React from 'react'

const Footer = () => {
  return (
    <footer className="page-footer green darken-2 center-align">
        Politico Website Designed by Okwara Emmanuel &copy; 2019
		</footer>
  )
}

export default Footer;