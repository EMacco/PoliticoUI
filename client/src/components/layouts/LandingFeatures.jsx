import React from 'react';
import electronicVote from '@base/img/electronic-vote.png';
import searchIcon from '@base/img/search-icon.png';
import readyIcon from '@base/img/ready-icon.png';
import calendarIcon from '@base/img/calendar-icon.png';
import petitionIcon from '@base/img/petition-icon.png';

const LandingFeatures = () => {
  return (
    <section className="container">
      <div className="app-feature-div" id="aboutUs">
        <img src={electronicVote} />
        <div className="feature-description">
          <h2>Vote</h2>
          <p>
            Politico allows voters to be able to cast their ballots using an internet connected
            device from anywhere in the world.
          </p>
        </div>
      </div>

      <hr />

      <div className="app-feature-div">
        <div className="feature-description">
          <h2>Search</h2>
          <p>
            Using Politico, you are able to view a complete list of Polical Parties as well as
            Candidates that are contesting for a particular office.
          </p>
        </div>
        <img src={searchIcon} />
      </div>

      <hr />

      <div className="app-feature-div">
        <img src={readyIcon} />
        <div className="feature-description">
          <h2>Express Interest</h2>
          <p>
            Are you planning to run for an office in the Government? <br />
            Politico allows you to easily declare interest against any Government Office.
          </p>
        </div>
      </div>

      <hr />

      <div className="app-feature-div">
        <div className="feature-description">
          <h2>Election Dates</h2>
          <p>With Poilitco, you can now view the dates of all pending and past elections.</p>
        </div>
        <img src={calendarIcon} />
      </div>

      <hr />

      <div className="app-feature-div">
        <img src={petitionIcon} />
        <div className="feature-description">
          <h2>Petitions</h2>
          <p>You can now create a petition against any concluded election.</p>
        </div>
      </div>
    </section>
  );
};

export default LandingFeatures;
