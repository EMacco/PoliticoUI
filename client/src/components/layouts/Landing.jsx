import React, { Fragment } from 'react';
import Banner from './Banner';
import Features from './LandingFeatures';
import Footer from './Footer';
import NavBar from '../layouts/NavBar';

const Landing = () => {
  return (
    <Fragment>
      <NavBar />
      |<main>
        <Banner />
        <Features />
        <Footer />
      </main>
    </Fragment>
  )
}

export default Landing;