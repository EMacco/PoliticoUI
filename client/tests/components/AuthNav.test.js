import React from 'react';
import { AuthNav } from '@components/layouts/AuthNav';

const props = {
  auth: {
    user: {
      firstname: '',
      lastname: '',
      phone: '',
      email: ''
    }
  },
  logoutUser: () => {}
};

const setup = () => {
  const wrapper = shallow(<AuthNav {...props} />);
  return wrapper;
};

describe('<AuthNav />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the Auth Navigation component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('AuthNav')).toBeTruthy();
  });
});
