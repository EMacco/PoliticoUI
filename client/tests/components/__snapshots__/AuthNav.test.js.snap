// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthNav /> should render the Auth Navigation component correctly 1`] = `
<section
  id="user-profile-container"
>
  <div>
    <img
      id="profile-img-preview"
    />
    <i>
      <label
        id="uploadingText"
      />
    </i>
  </div>
  <div
    className="profile-description-text"
  >
    <label>
      <span
        className="profile-answers"
        id="welcomeUserLbl2"
      >
         
      </span>
    </label>
    <label>
      Email:
       
      <span
        className="profile-answers"
        id="welcomeEmailLbl"
      />
    </label>
    <label>
      Phone:
       
      <span
        className="profile-answers"
        id="welcomePhoneLbl"
      />
    </label>
  </div>
  <hr
    className="separate-profile-nav"
  />
  <Link
    id={null}
    to="/dashboard"
  >
    <div
      className="nav-menu"
      id={null}
    >
      My Profile
    </div>
  </Link>
  <Link
    id={null}
    to="/parties"
  >
    <div
      className="nav-menu"
      id={null}
    >
      Political Parties
    </div>
  </Link>
  <Link
    id={null}
    to="/offices"
  >
    <div
      className="nav-menu"
      id={null}
    >
      Government Offices
    </div>
  </Link>
  <Link
    onClick={[Function]}
    to="/dashboard"
  >
    <div
      className="nav-menu"
    >
      Log out
    </div>
  </Link>
</section>
`;
